### PR TITLE
`[Move]` Part-3 Classes into Different Library - `Neo.Extensions`

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Description
Reorganized and move classes from `neo` library to `neo.io` and `neo.extensions` library. 

### Fixed problems and warnings that occurred. Along with naming violations.

# Change Log
 - Removed methods `BitLen` unused.
 - Move method `ToHexString` to `./src/Plugins/Neo.Extensions/ByteExtensions.cs`
 - Added `Using Neo.Extensions` where needed
 - Replaced method `Concat` with `C#` `collections` concat.

## Type of change

- [x] Optimization (the change is only an optimization)
- [x] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- Unit Testing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Plugins/MPTTrie/Cryptography/MPTTrie/Trie.Find.cs [hunk 1]
```diff
@@ -47,7 +46,7 @@ private ReadOnlySpan<byte> Seek(ref Node node, ReadOnlySpan<byte> path, out Node
                             start = node;
                             return ReadOnlySpan<byte>.Empty;
                         }
-                        return Concat(path[..1], Seek(ref node.Children[path[0]], path[1..], out start));
+                        return new([.. path[..1], .. Seek(ref node.Children[path[0]], path[1..], out start)]);
```
Comment (Jim8y) [inline src/Plugins/MPTTrie/Cryptography/MPTTrie/Trie.Find.cs:49]: These type of change maybe should be done in another pr, since they are not related.

File: src/Neo/Helper.cs [hunk 2]
```diff
@@ -29,59 +27,6 @@ public static class Helper
     {
         private static readonly DateTime unixEpoch = new(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static int BitLen(int w)
```
Comment (Jim8y) [inline src/Neo/Helper.cs:33]: where is this being moved to?

File: src/Plugins/MPTTrie/Cryptography/MPTTrie/Trie.Find.cs (hunk 1 above)
Comment (cschuchardt88) [reply src/Plugins/MPTTrie/Cryptography/MPTTrie/Trie.Find.cs:49]: It is replacing `Helper.Concat` with `C#` Concat collection.

File: src/Neo/Helper.cs (hunk 2 above)
Comment (cschuchardt88) [reply src/Neo/Helper.cs:33]: no where, It was never used and plus it was set as `private`.

File: src/Plugins/MPTTrie/Cryptography/MPTTrie/Trie.Find.cs (hunk 1 above)
Comment (shargon) [reply src/Plugins/MPTTrie/Cryptography/MPTTrie/Trie.Find.cs:49]: Agree, I'm not sure if this is faster, this change is not related to this PR

Comment (cschuchardt88) [reply src/Plugins/MPTTrie/Cryptography/MPTTrie/Trie.Find.cs:49]: It should be faster it gets compiled in at build. I think it related, because I said `Fixed problems and warnings that occurred. Along with naming violations.`

**Review by Jim8y — APPROVED** [review]

File: src/Neo/Helper.cs [hunk 3]
```diff
@@ -29,59 +27,6 @@ public static class Helper
...
-                : (w < 1 << 29 ? (w < 1 << 28 ? 28 : 29) : (w < 1 << 30 ? 30 : 31)))));
-        }
-
-        /// <summary>
-        /// Concatenates the specified byte arrays.
-        /// </summary>
-        /// <param name="buffers">The byte arrays to concatenate.</param>
-        /// <returns>The concatenated byte array.</returns>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static byte[] Concat(params byte[][] buffers)
```
Comment (shargon) [inline src/Neo/Helper.cs:56]: I prefer to avoid the concat change, or at least prove that is faster

Comment (cschuchardt88) [reply src/Neo/Helper.cs:56]: Yes it performance better
```
Getting the best performance for constructing each collection type can be tricky. Simple solutions often waste both CPU and memory. Having a literal form allows for maximum flexibility from the compiler implementation to optimize the literal to produce at least as good a result as a user could provide, but with simple code. Very often the compiler will be able to do better, and the specification aims to allow the implementation large amounts of leeway in terms of implementation strategy to ensure this.
```

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-12.0/collection-expressions#summary

**Review by shargon — APPROVED** [review]

**Review by Jim8y — APPROVED** [review]
